pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -230,6 +248,11 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 		}
 	}
 
+	if err := c.reconcileMetricsService(ctx, pa); err != nil {
+		logger.Errorw("Error reconciling metrics service", zap.Error(err))
+		return err
+	}
```

Comment:
Not sure if it matters at all, but should we reconcile this at the beginning of the reconciliation method? After all it's a prereq for all of the scaling business.